### ISSUE TITLE
Improve _NLinear implementation to use less RAM

### ISSUE THREAD

**Issue by dholzmueller** [yandex-research/rtdl-num-embeddings#19]
I noticed that the forward pass of the _NLinear class in package/rtdl_num_embeddings.py uses significantly more GPU RAM than necessary (I got RAM overflows for 10K numerical features). I could improve RAM usage as well as runtime by replacing the line
```
x = (x[..., None, :] @ self.weight).squeeze(-2)
```
by
```
x = (x.transpose(0, 1) @ self.weight).transpose(0, 1)
```
which is equivalent. Explanation: The latter line transposes x to have shape [n_cont, batch_size, emb_hidden], uses batch matrix multiplication with the weight of shape [n_cont, emb_hidden, emb_out], and then transposes back to have [batch_size, n_cont, emb_out]. Perhaps the first version has a less efficient implementation because it uses 4D batch matrix multiplication instead of 3D batch matrix multiplication.

**Comment by Yura52** [comment]
Thank you very much for sharing! I was never satisfied with the efficiency of `_NLinear`, but I thought if I used the native PyTorch matmul then PyTorch would automatically choose the most efficient implementation for given shapes. Indeed after quick testing of the suggested approach I see a noticeable efficiency improvement. I will keep the issue open for now to make it more visible, and I will close it when I update the package.

**Comment by Yura52** [comment]
The suggested change is implemented in the version 0.0.10 of the package ([release notes](https://github.com/yandex-research/rtdl-num-embeddings/releases/tag/v0.0.10)). Thanks!

**Comment by Yura52** [comment]
The suggested change is implemented in the version 0.0.10 of the package ([release notes](https://github.com/yandex-research/rtdl-num-embeddings/releases/tag/v0.0.10)). Thanks!
